botocore/utils.py

```diff
@@ -393,8 +393,10 @@ def __init__(
 
         if env is None:
             env = os.environ.copy()
-        self._disabled = env.get('AWS_EC2_METADATA_DISABLED', 'false').lower()
-        self._disabled = self._disabled == 'true'
+        self._disabled = (
+            env.get('AWS_EC2_METADATA_DISABLED', 'false').lower() == 'true'
+        )
```

Comment:
I agree that setting an instance var only to immediately change it is probably bad practice. I'm guessing it was done to reduce the number of lines. We can still do that, improve the quality and utilize something more reusable if we di this instead.
```suggestion
        raw_disabled = env.get('AWS_EC2_METADATA_DISABLED', 'false')
        self._disabled = ensure_boolean(raw_disabled)
```
At the very least we should use `ensure_boolean` which lowercases string inputs already.

Reply:
Ideally this should have never been a boolean, and enum for the mode makes more sense long term. I left feedback on the specification but that ship has probably already sailed.